images/builders/mlflow/run.sh

```diff
@@ -1,4 +1,57 @@
 #!/bin/sh
 
-mkdir -p .fuseml
-cp -rv ${FUSEML_FILES}/* .fuseml/ | awk '/Dockerfile/ {print $3}' | sed -e "s/[\r\n']//g"
+set -o pipefail
+
+# generates a checksum based on the contents of a conda.yaml file
+get_tag() {
+    file=${1:-"conda.yaml"}
+    dep_provider=""
+
+    while IFS= read -r line; do
+        name=$(echo ${line} | sed 's/ *- //g')
+        case $line in
+            '  - '*':')
+                dep_provider=$(printf '%s' "${name}" | tr -d :)
+            ;;
+            '  - '*)
+                if [ -z "${dependencies}" ]; then
+                    dependencies=${name}
+                else
+                    dependencies=${dependencies}'\n'${name}
+                fi
+            ;;
+            '    - '*)
+                dependencies=${dependencies}'\n'${dep_provider}'.'${name}
+            ;;
+        esac
+    done <"${file}"
+
+    printf "$dependencies" | sort | cksum | cut -f 1 -d ' '
```

Comment:
If you hash the `conda.yaml` a change in the package list order would be enough to invalidate the image.

Reply:
ok, but how often does that happen ? on the other hand, hashing the whole file is something simpler that could even be reused (e.g. with `requirements.txt` files). I would go for the simpler but more generic solution, even if there are some corner cases that aren't covered.

Reply:
depending on the use-case it can happen very often, for example when sharing the image between different codesets with the same requirements.

Considering that `requirements.txt` is just a list of packages adding a condition to support it is quite simple.

Reply:
> depending on the use-case it can happen very often, for example when sharing the image between different codesets with the same requirements.

I don't see that happening either, but if you're so sure about it, I'll take it